### PR TITLE
fix: qrcode fgColor for dark mode

### PR DESCRIPTION
**Pull request by ds1371dani**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

image

QRCode remains black in **dark** theme.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | :lipstick: Fix QRCode color in dark theme |
| 🇨🇳 Chinese | :lipstick: 修复深色主题下 QRCode 颜色问题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 00d333f</samp>

Improved the QRCode component's color handling. Made it possible to omit the `color` prop and use the theme's text color instead.

### <samp>🤖 Generated by Copilot at 00d333f</samp>

* Make the `color` prop of the QRCode component optional and use the theme token colorText as the default foreground color of the QR code ([link](https://github.com/ant-design/ant-design/pull/43162/files?diff=unified&w=0#diff-99175e9424e53dc9d08dd9c77c57c2ac26a0aff2f9b4eb0f6cf34d1709ab5662L24-R24), [link](https://github.com/ant-design/ant-design/pull/43162/files?diff=unified&w=0#diff-99175e9424e53dc9d08dd9c77c57c2ac26a0aff2f9b4eb0f6cf34d1709ab5662L54-R54))
* Update the `fgColor` option of the QRCodeGenerator to respect the `color` prop or the theme token colorText ([link](https://github.com/ant-design/ant-design/pull/43162/files?diff=unified&w=0#diff-99175e9424e53dc9d08dd9c77c57c2ac26a0aff2f9b4eb0f6cf34d1709ab5662L54-R54))

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
图片

File: components/qrcode/index.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ const QRCode: React.FC<QRCodeProps> = (props) => {
     size: size - (token.paddingSM + token.lineWidth) * 2,
     level: errorLevel,
     bgColor,
-    fgColor: color,
+    fgColor: color || token.colorText,
```
Comment (li-jia-nan) [inline components/qrcode/index.tsx:54]: 这个感觉可以直接用参数默认值

Comment (ds1371dani) [reply components/qrcode/index.tsx:54]: Yes you're correct, it would be better to move it to where all default values are, but it's been merged

Comment (li-jia-nan) [reply components/qrcode/index.tsx:54]: You can open a new PR.

Comment (ds1371dani) [reply components/qrcode/index.tsx:54]: new PR: https://github.com/ant-design/ant-design/pull/43164

Comment (li-jia-nan) [reply components/qrcode/index.tsx:54]: merged

**Pull request merged**